core/drivers/stm32_hpdma.c

```diff
@@ -168,12 +171,30 @@ static TEE_Result apply_rif_config(struct hpdma_pdata *hpdma_d, bool is_tdcid)
 		goto end;
 
 	for (i = 0; i < HPDMA_RIF_CHANNELS; i++) {
+		uint32_t cid_conf = hpdma_d->conf_data->cid_confs[i];
+
 		if (!(BIT(i) & hpdma_d->conf_data->access_mask[0]))
 			continue;
 
+		/*
+		 * Errata: When CID filtering is enabled on one of RISAB 3/4/5
+		 * instances, we forbid the use of CID0 for any initiator on the
+		 * bus to handle transient CID0 transactions on these RAMs.
+		 */
+		if (hpdma_d->errata_ahbrisab &&
+		    (!(cid_conf & _CIDCFGR_CFEN) ||
+		     (!(cid_conf & _CIDCFGR_SEMEN) &&
+		      ((cid_conf & _HPDMA_CIDCFGR_SCID_MASK) >>
+		       _HPDMA_CIDCFGR_SCID_SHIFT) == RIF_CID0) ||
+		     (cid_conf & _CIDCFGR_SEMEN &&
+		      cid_conf & BIT(_HPDMA_CIDCFGR_SEMWL_SHIFT)))) {
```

Comment:
Here also, prefer with an explicit reference to  `RIF_CID0`:
`		      cid_conf & BIT(RIF_CID0 + _HPDMA_CIDCFGR_SEMWL_SHIFT)))) {`